internal/commands/app/delete_test.go

```diff
@@ -0,0 +1,112 @@
+package app
+
+import (
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/utils/test/assert"
+	"github.com/10gen/realm-cli/internal/utils/test/mock"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+func TestAppDeleteHandler(t *testing.T) {
+	groupID1 := primitive.NewObjectID().Hex()
+	appID := "60344735b37e3733de2adf40"
+	app1 := realm.App{
+		ID:          appID,
+		GroupID:     groupID1,
+		ClientAppID: "app1-abcde",
+		Name:        "app1",
+	}
+
+	for _, tc := range []struct {
+		description    string
+		inputs         deleteInputs
+		apps           []realm.App
+		deleteErr      error
+		expectedApps   []string
+		expectedOutput string
+	}{
+		{
+			description:    "should delete no apps if none are found",
+			inputs:         deleteInputs{},
+			expectedOutput: "01:23:45 UTC INFO  No apps to delete\n",
+		},
+		{
+			description:  "with no project flag set and an apps flag set should delete all apps that match the apps flag",
+			inputs:       deleteInputs{Apps: []string{"app1"}},
+			apps:         []realm.App{app1},
```

Comment:
if im understanding the test correctly, might be worth making making an `app2` and including it here.

looking at the next test, maybe even a 3rd test app where:
* app1: groupID1, Name: app1
* app2: groupID1, Name: app2
* app3: groupID2, Name: app1

That way you can prove apps of the wrong name are filtered out here, and apps of the wrong Project are filtered out in the test below

Reply:
Really good idea here. Done!